src/jinjanator/formats.py

```diff
@@ -248,12 +250,8 @@ def parse(self, data_string: str) -> Mapping[str, str]:
         $ j2 config.j2 - < data.env
         """
 
-        return dict(
-            filter(
-                lambda line: len(line) == 2,  # noqa: PLR2004
-                (list(map(str.strip, line.split("=", 1))) for line in data_string.split("\n")),
-            ),
-        )
+        data = StringIO(data_string)
+        return dotenv_values(stream=data)
```

Comment:
There's a bit of an issue here; the return type of this function is `Mapping[str, str]`, but `dotenv_values` returns `Dict[str, Optional[str]]`. (this can also be see by running `hatch run lint:lint`).

While it may be safe to do this if Jinja2 can handle `Optional[str]`, it will require changing the definition of what a `format` plugin returns in the jinjanator-plugins repository and touching the other format plugins as well. Since the only benefit I can see from this would be allowing for variables with empty values, I'm not keen on making a change like that.

Would you consider using a dict comprehension here to remove any variables that don't have values, or actually giving them 'empty string' values?

Reply:
Updated to return an empty string as that more closely matched the previous behavior. Thank you for the feedback!